pkg/operator/sync.go

```diff
@@ -112,13 +112,18 @@ func isCloudConfigRequired(infra *configv1.Infrastructure) bool {
 	if infra.Spec.CloudConfig.Name != "" {
 		return true
 	}
-	switch infra.Status.PlatformStatus.Type {
-	case configv1.AzurePlatformType,
-		configv1.GCPPlatformType,
-		configv1.OpenStackPlatformType,
-		configv1.VSpherePlatformType:
-		return true
-	default:
+
+	if infra.Status.PlatformStatus != nil {
+		switch infra.Status.PlatformStatus.Type {
+		case configv1.AzurePlatformType,
+			configv1.GCPPlatformType,
+			configv1.OpenStackPlatformType,
+			configv1.VSpherePlatformType:
+			return true
+		default:
+			return false
+		}
+	} else {
```

Comment:
a nest can be removed here and above with:
```
if cfg.Infra.Status.PlatformStatus == nil {
    return ...
}
switch cfg.Infra.Status.PlatformStatus.Type {
...
```
Thinking twice on this, the change in `lib/resourcemerge/machineconfig.go` LGTM but following the trails for the panic indicates that our code isn't that robust to changes (and we're human and miss things in reviews, all good ofc!).
The original change w/o the nil check is here https://github.com/openshift/machine-config-operator/pull/1658/commits/e7455dcb4e0150e00f78e0ae4954b73047d1bf75#diff-fa45321336db7ad1cedc28bf643a4f97R107
Then, here's another (!!!) https://github.com/openshift/machine-config-operator/blob/master/pkg/operator/sync.go#L322 (in master, doesn't affect 4.5, but as we can notice, we aren't checking for nil there)
Then we can instead notice https://github.com/openshift/machine-config-operator/blob/master/pkg/operator/render.go#L80-L91 within `createDiscoveredControllerConfigSpec` which is the first thing we do after grabbing the infra object. In those lines we make sure that if the PlatformStatus is nil, we "adjust" it for compatibility reasons and then we store the infra object in the ControllerConfig object (there's another bug there tho, unrelated, we're modifying an object from the indexers which is a no-no).
Anyway, it seems that after grabbing the initial infra object (https://github.com/openshift/machine-config-operator/blob/master/pkg/operator/sync.go#L228), we do run it through a compatibility fix to make sure PlatformStatus is never nil from there (https://github.com/openshift/machine-config-operator/blob/master/pkg/operator/render.go#L80-L91), perhaps a forward looking fix would be to:

- grab the infra object (we do this already)
- run it through `createDiscoveredControllerConfigSpec` (we do this already)
- pass the resulting `ControllerConfig.Spec.Infra...` object in funcs like this and `getIgnitionHost`, instead of the `infra` global one, as we're sure that the infra in the CC already went through this very same nil check (we need to do this)

The above allows us to avoid adding specific nil checks for the same thing and help us tomorrow when the next PR adds something that access PlatformStatus w/o checking it